### PR TITLE
ddl: Speed up unit tests

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve? 
Speed up unit tests.
Before this PR:
`ok  	github.com/pingcap/tidb/ddl	25.541s	coverage: 81.1% of statements`
After this PR:
`ok   github.com/pingcap/tidb/ddl 13.158s coverage: 81.0% of statements`
### What is changed and how it works?

- Adjust the parallelism of the tests.

- Adjust the value of `SchemaOutOfDateRetryInterval` to reduce the time to possibly retry.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @winkyao @crazycs520 @bb7133

**Comment by zimulala** [comment]
PTAL @winkyao @crazycs520 @bb7133

File: ddl/db_integration_test.go [hunk 1]
```diff
@@ -98,8 +92,10 @@ func tearDownIntegrationSuiteTest(s *testIntegrationSuite, c *C) {
 	tk := testkit.NewTestKit(c, s.store)
 	tk.MustExec("use test")
 	r := tk.MustQuery("show tables")
+	tbls := make([]string, 0, len(r.Rows()))
```
Comment (bb7133) [inline ddl/db_integration_test.go]: seems this is redundant, right?

File: ddl/ddl.go [hunk 2]
```diff
@@ -472,12 +472,12 @@ func (d *ddl) close() {
 	for _, worker := range d.workers {
 		worker.close()
 	}
-	if d.sessPool != nil {
-		d.sessPool.close()
-	}
 	if d.delRangeMgr != nil {
 		d.delRangeMgr.clear()
 	}
+	if d.sessPool != nil {
```
Comment (bb7133) [inline ddl/ddl.go]: Is there a specific reason to move it here?

**Comment by zimulala** [comment]
PTAL @crazycs520 @bb7133

**Pull request merged**